### PR TITLE
ECS tags

### PR DESCRIPTION
**Pull request by metalgearsloth**
Requires https://github.com/space-wizards/RobustToolbox/pull/2513

### CODE REVIEW

File: Content.Shared/Tag/TagComponent.cs [hunk 1]
```diff
@@ -1,289 +1,15 @@
...
-    public class TagComponent : Component, ISerializationHooks
+    [RegisterComponent, Friend(typeof(TagSystem))]
+    public sealed class TagComponent : Component, ISerializationHooks
     {
         [ViewVariables]
         [DataField("tags", customTypeSerializer: typeof(PrototypeIdHashSetSerializer<TagPrototype>))]
-        private readonly HashSet<string> _tags = new();
+        public readonly HashSet<string> _tags = new();
 
         public IReadOnlySet<string> Tags => _tags;
```
Comment (mirrorcult) [inline Content.Shared/Tag/TagComponent.cs:13]: sus

File: Content.Shared/Tag/TagSystem.cs [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+            var added = component._tags.Add(id);
+
+            if (added)
+            {
+                Dirty(component);
+                return true;
+            }
+
+            return false;
+        }
```
Comment (mirrorcult) [inline Content.Shared/Tag/TagSystem.cs:310]: uhhh indentation randomly changes here

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by moonheart08** [comment]
Tags are common enough ideally there'd be something as easy as TryComp & co to use but i'm not sure that's not a pipedream

File: Content.Shared/Tag/TagSystem.cs (hunk 2 above)
Comment (metalgearsloth) [reply Content.Shared/Tag/TagSystem.cs:310]: copy-paste moment

**Pull request merged**